src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_button_press_notification.cc

```diff
@@ -52,155 +53,35 @@ OnButtonPressNotification::OnButtonPressNotification(
     rpc_service::RPCService& rpc_service,
     HMICapabilities& hmi_capabilities,
     policy::PolicyHandlerInterface& policy_handler)
-    : CommandNotificationImpl(message,
-                              application_manager,
-                              rpc_service,
-                              hmi_capabilities,
-                              policy_handler) {}
+    : ButtonNotificationToMobile(message,
+                                 application_manager,
+                                 rpc_service,
+                                 hmi_capabilities,
+                                 policy_handler) {}
 
 OnButtonPressNotification::~OnButtonPressNotification() {}
 
-void OnButtonPressNotification::Run() {
+void OnButtonPressNotification::SendButtonNotification(
+    ApplicationSharedPtr app) {
   SDL_LOG_AUTO_TRACE();
 
-  const uint32_t btn_id = static_cast<uint32_t>(
-      (*message_)[strings::msg_params][hmi_response::button_name].asInt());
+  mobile_apis::ButtonPressMode::eType btn_press_mode;
+  if ((*message_)[strings::msg_params].keyExists(hmi_response::button_mode)) {
```

Comment:
Is this created from both mobile and HMI requests? Do we need to check for both the HMI and mobile API parameter names?

also, why not include this logic in `CreateButtonNotificationToMobile`? Similar logic for `buttonName` is included there.

Reply:
@iCollin same as above. In this case, it is not so simple to generalize `buttonMode` because this parameter has different names in `OnButtonEvent` and `OnButtonPress` (however `buttonName` name is the same).
I moved this logic to `CreateButtonNotificationToMobile` by considering function id and customizing behavior.
See https://github.com/smartdevicelink/sdl_core/pull/3743/commits/16142947000f2ed54279306468a6c4cb1594805f